OpenRA.Mods.Cnc/Traits/MadTank.cs

```diff
@@ -92,8 +106,8 @@ public IEnumerable<IOrderTargeter> Orders
 		{
 			get
 			{
-				yield return new TargetTypeOrderTargeter(new BitSet<TargetableType>("DetonateAttack"), "DetonateAttack", 5, "attack", true, false) { ForceAttack = false };
-				yield return new DeployOrderTargeter("Detonate", 5);
+				yield return new TargetTypeOrderTargeter(new BitSet<TargetableType>("DetonateAttack"), "DetonateAttack", 5, info.AttackCursor, true, false) { ForceAttack = false };
+				yield return new DeployOrderTargeter("Detonate", 5, () => initiated ? info.DeployBlockedCursor : info.DeployCursor);
```

Comment:
Showing the standard deploy cursor after deploying was indeed bogus, so good spotting here.

IMO showing a blocked cursor is the wrong solution though, since these cursors usually indicate that something is generally possible but isn't right now. The MAD tank deployment is a one-way transformation, so I don't think we should be showing the deploy cursor as an option at all (equivalent to disabling the trait, if this code were rewritten properly).

This can be done using
```csharp
if (!initiated)
	yield return new DeployOrderTargeter("Detonate", 5, () => info.DeployCursor);
```
and then we could drop the `DeployBlockedCursor` definition.

Reply:
Oh yes nice, that looks way more polished. Thanks!